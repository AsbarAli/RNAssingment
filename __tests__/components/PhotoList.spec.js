import React from 'react';
import {shallow} from 'enzyme';

import PhotoListItem from '../../src/module/components/photoListItem/PhotoListItem.component';
import {checkProps} from '../../Utils';

describe('Photo List Item', () => {
  describe('Rendering', () => {
    const mockOnPress = jest.fn();
    const photoListItem = shallow(
      <PhotoListItem
        item={{thumbnailUrl: 'some'}}
        onPhotoClick={mockOnPress}
      />
    );

    it('should match to snapshot', () => {
      expect(photoListItem).toMatchSnapshot();
    });

    it('should render without errors', () => {
      expect(photoListItem.length).toBe(1);
    });
  });

  describe('Interaction', () => {
    describe('handle Photo Press', () => {
      const mockOnPress = jest.fn();
      let instance;

      beforeEach(() => {
        instance = shallow(
          <PhotoListItem
            item={{thumbnailUrl: 'some'}}
            onPhotoClick={mockOnPress}
          />
        ).instance();
        jest.clearAllMocks();
      });

      it('should call onPress', () => {
        instance.handlePhotoPress();

        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('Prop types', () => {
    describe('Checking Prop types', () => {
      it('it should not throw warning', () => {
        const mockOnPress = jest.fn();

        const expectedPropTypes = {item: {thumbnailUrl: 'http://something', url: 'http://blahblab'}, onPhotoClick: mockOnPress};

        const propsError = checkProps(PhotoListItem, expectedPropTypes);
        expect(propsError).toBeUndefined();
      });
    });
  });
});

