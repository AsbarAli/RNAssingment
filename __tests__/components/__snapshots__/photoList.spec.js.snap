// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo List Item Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
